packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -813,6 +813,276 @@ class _TextSelectionHandleOverlayState
+
+  /// Selects text position specified in [details] if selection is enabled.
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onSingleLongTapStart], which triggers this callback.
+  @protected
+  void onSingleLongTapStart(LongPressStartDetails details) {
+    if (client.selectionEnabled) {
+      renderEditable.selectPositionAt(
+        from: details.globalPosition,
+        cause: SelectionChangedCause.longPress,
+      );
+    }
+  }
+
+  /// Updates the selection location specified in [details] if selection is enabled.
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onSingleLongTapMoveUpdate], which triggers this callback.
+  @protected
+  void onSingleLongTapMoveUpdate(LongPressMoveUpdateDetails details) {
+    if (client.selectionEnabled) {
+      renderEditable.selectPositionAt(
+        from: details.globalPosition,
+        cause: SelectionChangedCause.longPress,
+      );
+    }
+  }
+
+  /// Shows tool bar if necessary.
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onSingleLongTapEnd], which triggers this callback.
+  @protected
+  void onSingleLongTapEnd(LongPressEndDetails details) {
+    if (_shouldShowSelectionToolbar)
+      editableText.showToolbar();
+  }
+
+  /// Selects a word through [renderEditable.selectWord] if selectionEnabled and
+  /// shows tool bar if necessary.
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onDoubleTapDown], which triggers this callback.
+  @protected
+  void onDoubleTapDown(TapDownDetails details) {
+    if (client.selectionEnabled) {
+      renderEditable.selectWord(cause: SelectionChangedCause.tap);
+      if (_shouldShowSelectionToolbar)
+        editableText.showToolbar();
+    }
+  }
+
+  /// Selects a text position specified in [details].
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onDragSelectionStart], which triggers this callback.
+  @protected
+  void onDragSelectionStart(DragStartDetails details) {
+    renderEditable.selectPositionAt(
+      from: details.globalPosition,
+      cause: SelectionChangedCause.drag,
+    );
+  }
+
+  /// Updates the selection location specified in [details].
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onDragSelectionUpdate], which triggers this callback.
+  @protected
+  void onDragSelectionUpdate(DragStartDetails startDetails, DragUpdateDetails updateDetails) {
+    renderEditable.selectPositionAt(
+      from: startDetails.globalPosition,
+      to: updateDetails.globalPosition,
+      cause: SelectionChangedCause.drag,
+    );
+  }
+
+  /// A place holder callback to enable subclass override.
+  ///
+  /// See also:
+  ///
+  ///  * [TextSelectionGestureDetector.onDragSelectionEnd], which triggers this callback.
+  @protected
+  void onDragSelectionEnd(DragEndDetails details) {/* Subclass should override this method if needed. */}
+
+  /// Build a [TextSelectionGestureDetector] with this handler provider.
```

Comment:
Returns a [TextSelectionGestureDetector] configured with the handlers provided by this builder.